shell/server/proxy.js

```diff
@@ -129,38 +130,68 @@ Meteor.methods({
     // running.
 
     check(grainId, String);
-
-    var sessionId = Random.id();
-    var user = Meteor.user();
-    var userId = user ? user._id : undefined;
-
-    // Start the grain if it is not running.
-    var runningGrain = runningGrains[grainId];
-    var grainInfo;
-    if (runningGrain) {
-      grainInfo = waitPromise(runningGrain);
-    } else {
-      grainInfo = continueGrain(grainId);
+    var grain = Grains.findOne(grainId);
+    if (!grain) {
+      throw new Meteor.Error(403, "Unauthorized", "No grain found.");
```

Comment:
Are we intentionally trying to hide from the user whether the grain exists? I'm not sure if that's really necessary, so I'd probably make this 404.